pkg/antctl/raw/traceflow/command.go

```diff
@@ -0,0 +1,337 @@
+	clientset "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+)
+
+var (
+	Command     *cobra.Command
+	source      string
+	destination string
+	outputType  string
+	waiting     bool
+	flow        string
+)
+
+// Response is the response of antctl traceflow.
+type Response struct {
+	Name        string                  `json:"name" yaml:"name"`      // Traceflow name
+	Phase       v1alpha1.TraceflowPhase `json:"phase,omitempty"`       // Traceflow phase
+	Source      string                  `json:"source,omitempty"`      // Traceflow source, e.g. "default/pod0"
+	Destination string                  `json:"destination,omitempty"` // Traceflow destination, e.g. "default/pod1"
+	NodeResults []v1alpha1.NodeResult   `json:"results,omitempty"`     // Traceflow node results
+}
+
+func init() {
+	Command = &cobra.Command{
+		Use:     "traceflow",
+		Short:   "Start a Traceflows",
+		Long:    "Start a Traceflows from one pod to another.",
+		Aliases: []string{"tf", "traceflows"},
+		Example: `  Start a Traceflow from busybox0 to busybox1, both pods are in namespace default
+  $antctl traceflow -S busybox0 -D busybox1
+  Start a Traceflow from busybox0 to destination IP, source is in namespace default
+  $antctl traceflow -S busybox0 -D 123.123.123.123
+  Start a Traceflow from busybox0 in namespace ns0 to busybox1 in namespace ns1, output type is json
+  $antctl traceflow -S ns0/busybox0 -D ns1/busybox1 -o json
+  Start a Traceflow from busybox0 to busybox1, with TCP header and 80 as destination port
+  $antctl traceflow -S busybox0 -D busybox1 -f tcp,tcp_dst=80
+`,
+		RunE: runE,
+	}
+
+	Command.Flags().StringVarP(&source, "source", "S", "", "source of the traceflow: namespace/pod or pod")
+	Command.Flags().StringVarP(&destination, "destination", "D", "", "destination of the traceflow: namespace/pod, pod or IP")
+	Command.Flags().StringVarP(&outputType, "output", "o", "yaml", "output type: yaml (default), json")
+	Command.Flags().BoolVarP(&waiting, "wait", "", true, "if false, command returns without retrieving results")
+	Command.Flags().StringVarP(&flow, "flow", "f", "", "specify the flow (packet headers) of the traceflow packet")
+}
+
+func runE(cmd *cobra.Command, _ []string) error {
+	if len(source) == 0 || len(destination) == 0 {
+		fmt.Println("Please provide source and destination.")
+		return nil
+	}
+
+	kubeconfigPath, err := cmd.Flags().GetString("kubeconfig")
+	if err != nil {
+		return err
+	}
+	kubeconfig, err := runtime.ResolveKubeconfig(kubeconfigPath)
+	if err != nil {
+		return err
+	}
+	setupKubeconfig(kubeconfig, &v1alpha1.SchemeGroupVersion)
+
+	client, err := clientset.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("error when creating clientset: %w", err)
+	}
+
+	tf, err := newTraceflow()
+	if err != nil {
+		return fmt.Errorf("error when filling up traceflow config: %w", err)
+	}
+
+	if _, err = client.OpsV1alpha1().Traceflows().Create(context.TODO(), tf, v1.CreateOptions{}); err != nil {
```

Comment:
I think it is better to use a timeout context here, since the network may not always work.

Reply:
Updated.